### PR TITLE
Add doc in types.go for hooks/description.sh

### PR DESCRIPTION
**Pull request by derekwaynecarr**
@bgrant0607 - the list of omissions reported in #4542 and some extras are added here.

I still can't get hack/verify-description.sh to report clean yet, and I am still struggling with bash to see what the case is.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Fixing the last few now, bash finally worked for me ;-)

File: pkg/api/v1beta1/types.go [hunk 1]
```diff
@@ -445,8 +446,10 @@ const (
 
 // TODO: add LastTransitionTime, Reason, Message to match NodeCondition api.
 type PodCondition struct {
-	Kind   PodConditionKind `json:"kind" description:"kind of the condition, currently only Ready"`
-	Status ConditionStatus  `json:"status" description:"status of the condition, one of Full, None, Unknown"`
+	// Kind is the kind of the condition
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Do we need the comments, since we have the descriptions?

File: pkg/api/v1beta1/types.go [hunk 2]
```diff
@@ -710,17 +713,18 @@ type Namespace struct {
 	Labels map[string]string `json:"labels,omitempty" description:"map of string keys and values that can be used to organize and categorize namespaces"`
 
 	// Spec defines the behavior of the Namespace.
-	Spec NamespaceSpec `json:"spec,omitempty"`
+	Spec NamespaceSpec `json:"spec,omitempty" description:"spec defines the behavior of the Namespace"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: I usually put `specification of the desired state of the Foo`, but what you have is fine.

**Comment by derekwaynecarr** [comment]
Fixes #4542 

@smarterclayton @yujuhong @bgrant0607 - please review.

hack/verify-description.sh now reports a clean result for me with this.

File: pkg/api/v1beta1/types.go (hunk 2 above)
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Except that I'd eliminate the field name from the description.

File: pkg/api/v1beta1/types.go [hunk 3]
```diff
@@ -710,17 +713,18 @@ type Namespace struct {
...
 
 	// Status describes the current status of a Namespace
-	Status NamespaceStatus `json:"status,omitempty"`
+	Status NamespaceStatus `json:"status,omitempty" description:"status describes the current status of a Namespace"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: I would omit "status describes"

File: pkg/api/v1beta1/types.go (hunk 1 above)
Comment (derekwaynecarr) [inline pkg/api/v1beta1/types.go]: I just added for consistency.

File: pkg/api/v1beta1/types.go [hunk 4]
```diff
@@ -1017,17 +1021,17 @@ const (
...
 	// Min usage constraints on this kind by resource name
-	Min ResourceList `json:"min,omitempty"`
+	Min ResourceList `json:"min,omitempty" description:"min usage constraints on this kind by resource name"`
 }
 
 // LimitRangeSpec defines a min/max usage limit for resources that match on kind
 type LimitRangeSpec struct {
 	// Limits is the list of LimitRangeItem objects that are enforced
-	Limits []LimitRangeItem `json:"limits"`
+	Limits []LimitRangeItem `json:"limits" description:"limits is the list of LimitRangeItem objects that are enforced"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Omit "limits is the".

File: pkg/api/v1beta1/types.go (hunk 2 above)
Comment (derekwaynecarr) [inline pkg/api/v1beta1/types.go]: I don't have a ton more time to refine this today, so if the goal is to enable the script to not allow more deviation, then this should be good to go.  We can nit text later I guess.

File: pkg/api/v1beta1/types.go [hunk 5]
```diff
@@ -1035,7 +1039,7 @@ type LimitRange struct {
 	TypeMeta `json:",inline"`
 
 	// Spec defines the limits enforced
-	Spec LimitRangeSpec `json:"spec,omitempty"`
+	Spec LimitRangeSpec `json:"spec,omitempty" description:"spec defines the limits enforced"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: How about "resource limits enforced"

File: pkg/api/v1beta1/types.go [hunk 6]
```diff
@@ -1067,9 +1071,9 @@ type ResourceQuotaSpec struct {
 // ResourceQuotaStatus defines the enforced hard limits and observed use
 type ResourceQuotaStatus struct {
 	// Hard is the set of enforced hard limits for each named resource
-	Hard ResourceList `json:"hard,omitempty"`
+	Hard ResourceList `json:"hard,omitempty" description:"hard is the set of enforced hard limits for each named resource"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Omit "hard is the"

File: pkg/api/v1beta1/types.go [hunk 7]
```diff
@@ -1067,9 +1071,9 @@ type ResourceQuotaSpec struct {
...
 	// Used is the current observed total usage of the resource in the namespace
-	Used ResourceList `json:"used,omitempty"`
+	Used ResourceList `json:"used,omitempty" description:"used is the current observed total usage of the resource in the namespace"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Omit "used is the"

Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Anyway, you get the idea. Omit the field name, "is", and "the". 

"Defines"/"specifies" and "describes" might sometimes be appropriate to distinguish spec and status, but might not always be necessary.

File: pkg/api/v1beta1/types.go [hunk 8]
```diff
@@ -158,7 +158,7 @@ type GitRepo struct {
 // Adapts a Secret into a VolumeSource
 type SecretSource struct {
 	// Reference to a Secret
-	Target ObjectReference `json:"target"`
+	Target ObjectReference `json:"target" description:"target is a reference to a secret"`
```
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Thanks. We'll merge the PR to fix the Travis check right after we merge this PR so more of these don't get merged.

Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Just "reference to a secret object"

File: pkg/api/v1beta1/types.go (hunk 1 above)
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Ok, though I don't think we're very consistent about it, and it just creates 2 things to update and keep in sync.

**Comment by bgrant0607** [comment]
Thanks much. For consistency and concision, I'd prefer the descriptions to be a little more terse. Swagger is going to provide the field names, so I definitely think we don't need to duplicate those.

File: pkg/api/v1beta1/types.go (hunk 2 above)
Comment (bgrant0607) [inline pkg/api/v1beta1/types.go]: Ok, fine.

**Comment by derekwaynecarr** [comment]
I can refine based on your comments Monday.  I just had to leave and get my kids from school and didn't want you to wait.

**Comment by bgrant0607** [comment]
It's fine. Thanks for the quick turnaround. Have a good weekend.

**Pull request merged**